### PR TITLE
Changed and fixed a few things

### PR DESCRIPTION
**Pull request by wzwietering**
-Out of bounds component placement on small screens
-Default scenario of the ScenarioFactory is the freeplay scenario
-Freeplay scenario doesn't have default components anymore
-Freeplay scenario is started when you finish the exercises
-Scenario 5 starts with an empty circuit
-Scenario 5 contains feedback for component count
-Scenario 5 informs the user that he completed all exercises at completion

### CODE REVIEW

**Comment by wzwietering** [comment]
-Scenario 3 requires resistance

File: app/src/main/java/com/edulectronics/tinycircuit/Models/Scenarios/ImplementedScenarios/Scenario3.java [hunk 1]
```diff
@@ -34,17 +35,20 @@ public boolean isCompleted(Circuit circuit) {
...
                     lampIsOn = ((Lightbulb) component).isOn;
                 } else if (component.getClass() == Switch.class) {
                     hasConnectedSwitch = component.hasOutputConnection(component.getConnectionPointByIndex(1));
+                } else if (component.getClass() == Resistor.class) {
+                    hasResistor = component.hasOutputConnection(component.getConnectionPointByIndex(1));
                 }
             }
         }
-        return (hasConnectedSwitch && lampIsOn);
+        return (hasConnectedSwitch && lampIsOn && hasResistor);
```
Comment (Jhsteensma) [inline app/src/main/java/com/edulectronics/tinycircuit/Models/Scenarios/ImplementedScenarios/Scenario3.java]: Moet misschien dan ook een Hint krijgen toch?

Comment (wzwietering) [inline app/src/main/java/com/edulectronics/tinycircuit/Models/Scenarios/ImplementedScenarios/Scenario3.java]: Die heeft hij ;)

**Comment by wzwietering** [comment]
-Component count checks for scenario 2 and scenario 3
-All scenarios require a full circle now to complete

**Pull request merged**